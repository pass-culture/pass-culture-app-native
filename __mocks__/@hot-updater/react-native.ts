import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
