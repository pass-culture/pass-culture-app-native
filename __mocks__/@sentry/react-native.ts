import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
