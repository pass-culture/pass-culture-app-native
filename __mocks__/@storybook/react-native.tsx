import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
