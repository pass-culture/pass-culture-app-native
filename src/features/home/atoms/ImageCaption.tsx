import React from 'react'
import { PixelRatio } from 'react-native'
import styled from 'styled-components/native'

import { ColorsEnum, Typo, MARGIN_DP, GUTTER_DP, getSpacing } from 'ui/theme'
import { BorderRadiusEnum } from 'ui/theme/grid'

interface ImageCaptionProps {
  category: string
  distance?: string
  imageWidth: number
}

export const ImageCaption = ({ category, imageWidth, distance }: ImageCaptionProps) => {
  return (
    <Row width={imageWidth}>
      <TextWrapper>
        <Typo.Caption color={ColorsEnum.WHITE} testID="categoryImageCaption">
          {category}
        </Typo.Caption>
      </TextWrapper>
      {distance && (
        <React.Fragment>
          <Separator />
          <TextWrapper>
            <Typo.Caption color={ColorsEnum.WHITE} testID="distanceImageCaption">
              {distance}
            </Typo.Caption>
          </TextWrapper>
        </React.Fragment>
      )}
    </Row>
  )
}

const rowHeight = PixelRatio.roundToNearestPixel(MARGIN_DP)
const textLineHeight = PixelRatio.roundToNearestPixel(GUTTER_DP)

const Row = styled.View<{ width: number }>(({ width }) => ({
  flexDirection: 'row',
  backgroundColor: ColorsEnum.BLACK,
  height: rowHeight,
  width,
  borderBottomLeftRadius: BorderRadiusEnum.BORDER_RADIUS,
  borderBottomRightRadius: BorderRadiusEnum.BORDER_RADIUS,
  alignItems: 'center',
}))

const Separator = styled.View({
  height: textLineHeight,
  backgroundColor: ColorsEnum.WHITE,
  width: 1,
})

const TextWrapper = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
  paddingHorizontal: getSpacing(1),
  flex: 1,
})
