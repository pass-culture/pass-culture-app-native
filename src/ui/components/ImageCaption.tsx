import React from 'react'
import { PixelRatio } from 'react-native'
import styled from 'styled-components/native'

import { Typo, GUTTER_DP, getSpacing } from 'ui/theme'

interface ImageCaptionProps {
  categoryLabel: string | null
  distance?: string
  height: number
  width: number
}

export const ImageCaption = ({ categoryLabel, height, width, distance }: ImageCaptionProps) => {
  return (
    <Row height={height} width={width}>
      <TextWrapper>
        <CategoryLabel numberOfLines={1} testID="categoryImageCaption">
          {categoryLabel}
        </CategoryLabel>
      </TextWrapper>
      {!!distance && (
        <React.Fragment>
          <Separator />
          <TextWrapper>
            <Distance testID="distanceImageCaption">{distance}</Distance>
          </TextWrapper>
        </React.Fragment>
      )}
    </Row>
  )
}

const CategoryLabel = styled(Typo.Caption)(({ theme }) => ({
  color: theme.colors.white,
}))

const Distance = styled(Typo.Caption)(({ theme }) => ({
  color: theme.colors.white,
}))

const textLineHeight = PixelRatio.roundToNearestPixel(GUTTER_DP)

const Row = styled.View<{ height: number; width: number }>(({ theme, height, width }) => ({
  flexDirection: 'row',
  backgroundColor: theme.colors.black,
  height,
  width,
  borderBottomLeftRadius: theme.borderRadius.radius,
  borderBottomRightRadius: theme.borderRadius.radius,
  alignItems: 'center',
}))

const Separator = styled.View(({ theme }) => ({
  height: textLineHeight,
  backgroundColor: theme.colors.white,
  width: 1,
}))

const TextWrapper = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
  // So all categories are entirely visibles on homepage offer tiles
  paddingHorizontal: getSpacing(0.5),
  flex: 1,
})
