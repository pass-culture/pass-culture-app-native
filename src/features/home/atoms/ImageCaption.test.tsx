import { render } from '@testing-library/react-native'
import React from 'react'

import { ImageCaption } from './ImageCaption'

let props = {
  category: 'Musique',
  distance: '1,2km',
  imageWidth: 50,
}

describe('ImageCaption component', () => {
  afterAll(() => jest.resetAllMocks())

  it('should render correctly', () => {
    const { toJSON } = render(<ImageCaption {...props} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should have an ellipsis', () => {
    const { getByTestId, queryByTestId } = render(<ImageCaption {...props} />)
    expect(getByTestId('categoryImageCaption').parent?.props.numberOfLines).toEqual(1)
    expect(queryByTestId('distanceImageCaption')).toBeTruthy()
  })

  it('should not display the distance if not available', () => {
    const { queryByTestId } = render(<ImageCaption {...props} distance={undefined} />)
    expect(queryByTestId('distanceImageCaption')).toBeNull()
  })
})
