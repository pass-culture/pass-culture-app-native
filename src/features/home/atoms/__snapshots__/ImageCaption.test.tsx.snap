// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCaption component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#151515",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "flexDirection": "row",
        "height": 24,
        "width": 50,
      },
    ]
  }
  width={50}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 8,
        },
      ]
    }
  >
    <Text
      color="#ffffff"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "600",
            "lineHeight": 16,
          },
        ]
      }
      testID="categoryImageCaption"
    >
      Musique
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "height": 16,
          "width": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 8,
        },
      ]
    }
  >
    <Text
      color="#ffffff"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "600",
            "lineHeight": 16,
          },
        ]
      }
      testID="distanceImageCaption"
    >
      <Text>
        1,2km
      </Text>
    </Text>
  </View>
</View>
`;
